### DISCUSSION TITLE
Getting an error.

### DISCUSSION THREAD

**Discussion by ayushg647** [mrdbourke/pytorch-deep-learning#488] [Q&A]
KeyError: 'You have to pass data to augmentations as named arguments, for example: aug(image=image)'

Got this error when implementing the statement "for X_test, y_test in test_dataloader:" in the testing loop., while making the computer vision model on the fashion dataset according to the tutorial.
How to rectify the error...

**Comment by mrdbourke** [comment]
Hi @ayushg647 , 

It's a bit hard to troubleshoot your error without seeing your code.

What have you tried so far? 

Have you tried to reference the code here: https://www.learnpytorch.io/03_pytorch_computer_vision/ 

You could compare your code to that in the notebook and see if you're doing the right steps.

**Reply by ayushg647** [reply]
Yes, I copy pasted the exact same code as in the notebook, but its showing this key error, asking for aug(image=image).

**Reply by mrdbourke** [reply]
Hmm, I'm not sure what might be happening here. 

I just ran all of the code in notebook 03 and it worked for me.

Do you mind sharing your code and showing me where the error is coming from?

**Reply by ayushg647** [reply]
# Import tqdm for progress bar
from tqdm.auto import tqdm

# Set the seed and start the timer
torch.manual_seed(42)
train_time_start_on_cpu = timer()

# Set the number of epochs (we'll keep this small for faster training times)
epochs = 3

# Create training and testing loop
for epoch in tqdm(range(epochs)):
    print(f"Epoch: {epoch}\n-------")
    ### Training
    train_loss = 0
    # Add a loop to loop through training batches
    for batch, (X, y) in enumerate(train_dataloader):
        model_0.train() 
        # 1. Forward pass
        y_pred = model_0(X)

        # 2. Calculate loss (per batch)
        loss = loss_fn(y_pred, y)
        train_loss += loss # accumulatively add up the loss per epoch 

        # 3. Optimizer zero grad
        optimizer.zero_grad()

        # 4. Loss backward
        loss.backward()

        # 5. Optimizer step
        optimizer.step()

        # Print out how many samples have been seen
        if batch % 400 == 0:
            print(f"Looked at {batch * len(X)}/{len(train_dataloader.dataset)} samples")

    # Divide total train loss by length of train dataloader (average loss per batch per epoch)
    train_loss /= len(train_dataloader)
    
    ### Testing
    # Setup variables for accumulatively adding up loss and accuracy 
    test_loss, test_acc = 0, 0 
    model_0.eval()
    with torch.inference_mode():
        for X_test, y_test in test_dataloader:
            # 1. Forward pass
            test_pred = model_0(X)
           
            # 2. Calculate loss (accumatively)
            test_loss += loss_fn(test_pred, y) # accumulatively add up the loss per epoch

            # 3. Calculate accuracy (preds need to be same as y_true)
            test_acc += accuracy_fn(y_true=y, y_pred=test_pred.argmax(dim=1))
        
        # Calculations on test metrics need to happen inside torch.inference_mode()
        # Divide total test loss by length of test dataloader (per batch)
        test_loss /= len(test_dataloader)

        # Divide total accuracy by length of test dataloader (per batch)
        test_acc /= len(test_dataloader)

    ## Print out what's happening
    print(f"\nTrain loss: {train_loss:.5f} | Test loss: {test_loss:.5f}, Test acc: {test_acc:.2f}%\n")

# Calculate training time      
train_time_end_on_cpu = timer()
total_train_time_model_0 = print_train_time(start=train_time_start_on_cpu, 
                                           end=train_time_end_on_cpu,
                                           device=str(next(model_0.parameters()).device))

**Reply by ayushg647** [reply]
for X_test, y_test in test_dataloader:
im getting an error in this line

**Reply by ayushg647** [reply]
Basically im not able to iterate on the test_loader

**Reply by 4833R11Y45** [reply]
did you find the fix?
